lightning-invoice/src/utils.rs

```diff
@@ -510,6 +540,52 @@ where
 	}
 }
 
+
+/// Used to store information about all the HTLCs that are inflight across all payment attempts
+pub(crate) struct AccountForInFlightHtlcs<'a, S: Score> {
+	scorer: &'a mut S,
+	/// Maps a channel's short channel id and its direction to the liquidity used up.
+	inflight_htlcs: HashMap<(u64, bool), u64>,
```

Comment:
Do we want to reuse the `InFlightHtlcs` structure here?

Reply:
I believe so, otherwise `InflightHtlcs::used_liquidity_msat` isn't actually called, IIUC. We'll need to define this in the `lightning` crate, however, given the dependency graph. May need to do so anyhow in the next follow-up, which will move tracking from `InvoicePayer` to `ChannelManager`.

Reply:
resolved in [24f9dc6](https://github.com/lightningdevkit/rust-lightning/pull/1694/commits/24f9dc638d408567f0197d7df1e1c6e8bd447d3c)

Reply:
Ah, I wasn't thinking straight on where this needed to be defined since it is used internally in `AccountForInFlightHtlcs` it can remain where you have it.